### PR TITLE
Implemented add-remove functionality to To-do list

### PR DESCRIPTION
**Pull request by mullenlucas**
## Implemented features

- Removed hard-coded previous list
- Created the functionality to add lists to the HTML and the Tasks class
- Once the list is populated, the functionality to edit and remove becomes available from the item's icon, showing 2 options:
  - Edit: edits the field both in HTML and on the Tasks class
  - Remove: removes an item from the to-do list and HTML
- All changes are recorded, followed in the indexes values, and stored in the local storage

### CODE REVIEW

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# INVALID ❌ ❌ ❌
# Hello @mullenlucas  👋  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, where adding a todo item feature isn't working and that's why we can't review other features. Please fix this so that we could completely review it. 👍🏻 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by mullenlucas** [comment]
As the previous submit was invalid, I'm not completely sure why.
For this new submit, I built the new html (in the `/dist` folder), as I understand that may be it, and made sure everything was working on different browsers.
Also, moved implemented functionalities to arrow functions

File: src/index.js [hunk 1]
```diff
@@ -1,54 +1,90 @@
...
+  currentIndex -= 1;
+  localStorage.setItem('todoItems', JSON.stringify(tasksCl.allTasks));
+  menuExp.style.display = 'none';
+});
+
+editTask.addEventListener('click', () => {
+  editHtml(numt, tasksCl);
+
+  menuExp.style.display = 'none';
+});
```
Comment (codecaiine) [inline src/index.js:100]: - [x] Well done with the edit function. However, when I press a key to edit everything disappears. I have to write my task  all over again Please correct that :+1:

File: src/index.js [hunk 2]
```diff
@@ -1,54 +1,90 @@
...
-  liB.appendChild(p);
-
-  const i = document.createElement('i');
-  i.setAttribute('class', 'fa-solid fa-ellipsis-vertical');
-  liB.appendChild(i);
-
-  todoUl.appendChild(liB);
-});
+let currentIndex = 0;
+let numt = 0;
```
Comment (codecaiine) [inline src/index.js:14]: - [x] From project requirement by default, new tasks should have the property `completed` set to `false` you did well :+1: But the property `index` set to the value of the new array length (i.e. if I have 4th tasks in my list, the index of that task should equal to 4 ) in your case if my task is 4th the index equal 3 :cry:  Please fix it to follow the requirement :+1: 
- [ ] Another thing I would like to mention is as required all hardcoded items from the tasks array should be removed. But in your case that action is impossible. Please fix it :smile:

Screenshot from 2022-08-12 21-15-20

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @mullenlucas , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Descriptive PR :heavy_check_mark: 
- To-Do List changes are saved in local storage :heavy_check_mark: 
- No linters errors :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (mullenlucas) [reply src/index.js:14]: @codecaiine 
Hello, thanks for all the suggestions, I will take a look at them now.

> * When I check a task the completed property needs to change from `false `to `true ` But in your case not even when I refresh the page. Please fix it 😄

One thing on the last one about the 'completed' properties, as I understand shouldn't that be for the next exercise ? If I'm not mistaken, this one is just for the add, remove and edit features

Comment (mullenlucas) [reply src/index.js:14]: @codecaiine 
> * Another thing I would like to mention is as required all hardcoded items from the tasks array should be removed. But in your case that action is impossible . Please fix it 😄

Sorry to bother, one more thing, I'm not sure about this point. I had some example items hardcoded previously which I removed, maybe I'm not understanding which parts are hard-coded and shouldn't be, could I ask for an example ? Thanks

Comment (codecaiine) [reply src/index.js:14]: Hi @mullenlucas, I just noticed it,  sorry you are right this is supposed to be for the following milestone :smile:  

Please make these changes  below and submit the next reviewer will check :+1: 

> * But the property `index` set to the value of the new array length (i.e. if I have 4th task in my list, the index of that task should equal to 4 ) in your case if my task is 4th the index equal 3 cry  Please fix it to follow the requirement +1

Comment (mullenlucas) [reply src/index.js:14]: > Hi @mullenlucas, I just noticed it, sorry you are right this is supposed to be for the following milestone 😄
> 
> Please make these changes below and submit the next reviewer will check 👍
> 
> > * But the property `index` set to the value of the new array length (i.e. if I have 4th task in my list, the index of that task should equal to 4 ) in your case if my task is 4th the index equal 3 cry  Please fix it to follow the requirement +1

Excellent, thank you

Comment (Reem-lab) [reply src/index.js:14]: - [x] As I can see nothing works as required I can't add any new task when I write a task and press on the icon or enter nothing is added to the list kindly check that and work on it to be fixed 👌🏻💯💯



https://user-images.githubusercontent.com/58553711/184453999-c0c39edd-3e64-42cc-a387-40188acc2b97.mp4

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @mullenlucas, 👋🏻👋🏻 

Great work on making the changes requested by a previous reviewer 👏🔥🔥🚀🚀 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights✨✨
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file. ✅


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
> * [OPTIONAL] I suggest adding the `MIT.md` file to your project :ok_hand:, Here is the [Micorverse Template](https://github.com/microverseinc/readme-template/blob/master/MIT.md), and remember to customize it :wink:
>
> > The MIT license gives users express permission to reuse code for any purpose, sometimes even if the code is part of proprietary software. As long as users include the original copy of the MIT license in their distribution, they can make any changes or modifications to the code to suit their own needs. :100:
>


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (mullenlucas) [reply src/index.js:14]: > * [ ]  As I can see nothing works as required I can't add any new task when I write a task and press on the icon or enter nothing is added to the list kindly check that and work on it to be fixed 👌🏻💯💯
> 
>  Project.Title.-.Google.Chrome.2022-08-13.00-44-16.mp4

Hello @Reem-lab , thank you for your revision

However I'm a little lost, previous reviewers could run those functionalities. I'm not sure how to proceed, I ran multiple tests on different browsers (Mozilla, Chrome and Edge) and they all seem to work.
I add a video of a demonstration on it working:

https://user-images.githubusercontent.com/69157785/184459354-8baa2a40-0e0c-4c1c-9738-5cedb95be755.mp4

I am not sure what might be happening, as I understand it is working, don't really know what to modify. 
Please let me know any suggestion on how to proceed

Regards,
Lucas

**Comment by mullenlucas** [comment]
Hello @Reem-lab 

I'm a little bit lost here because I don't know how to proceed. As I understand, the previous reviewers could run the functionalities shown on the video.
I also tried running the website on different browsers (Edge, Chrome and Firefox), and all of them seem to work. Here I append a video showing how it works:


https://user-images.githubusercontent.com/69157785/184462057-6c52732e-32f8-490b-b7f9-54fcfb972b03.mp4

I wouldn't know what to fix since it seems to be working, also my previous reviewers also had it working.
Please let me know any suggestions on how to proceed or what could I review

Regards,
Lucas

**Comment by Reem-lab** [comment]
Hi @mullenlucas I cloned your repo again to make sure, it doesn't work at all for me I don't know what the reason maybe you made something wrong after the first review, you can ask for another review 👍🏻

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @mullenlucas

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**